File: cmdchallenge/oops/src/oops.nim
import os
import argparse
import sequtils

let p = newParser("oops"):
  option("-t", "--timeout", help="how many seconds to sleep", default="10000")

var opts = p.parse

let timeout = parseInt(opts.timeout)
for _ in repeat(0, timeout):
  os.sleep(1000)

File: cmdchallenge/cmd/oops/oops.go
package main

import (
	"flag"
	"time"
)

const defaultTimeout int = 1000

func main() {
	timeout := flag.Int("timeout", defaultTimeout, "how many seconds to sleep")
	flag.Parse()

	for i := 0; ; i++ {
		if timeout != nil && i >= *timeout {
			break
		}
		time.Sleep(1 * time.Second)
	}
}
